### ISSUE TITLE
Не загружаются объёмные ISO-образы в хранилище

### ISSUE THREAD

**Issue by 2lifehd** [Aerodisk/openvair#23]
Не загружаются объёмные ISO-образы (больше 3Гб) в хранилище. Возможно это связано, что в корневом разделе всего 6Гб.
Заметил, что образ сначала копируется в раздел /tmp, после этого в корне системы остается меньше места, чем будет вторая копия образа. Возможна причина в этом?

**Comment by tihon49** [comment]
Да, вы правы. Сначала образ грузится в `/tmp`. Посмотрим как можно улучшить этот момент. Но не забывайте, что Вы тоже можете предложить вариант решения )

**Comment by space2pacman** [comment]
> Да, вы правы. Сначала образ грузится в `/tmp`. Посмотрим как можно улучшить этот момент. Но не забывайте, что Вы тоже можете предложить вариант решения )

А если сделать через NBD (Network Block Device)?
* Подключать ISO в браузере.
* Устанавливать соединение с NBD сервером.
* Сервер сам решает какие байты принимать.

Тем самым можно подключать ISO в неограниченном размере а на сервере не тратить места для временных файлов.

**Comment by 2lifehd** [comment]
Возможно это будет приемлемым решением.

**Comment by tihon49** [comment]
> > Да, вы правы. Сначала образ грузится в `/tmp`. Посмотрим как можно улучшить этот момент. Но не забывайте, что Вы тоже можете предложить вариант решения )
> 
> А если сделать через NBD (Network Block Device)?
> 
> * Подключать ISO в браузере.
> * Устанавливать соединение с NBD сервером.
> * Сервер сам решает какие байты принимать.
> 
> Тем самым можно подключать ISO в неограниченном размере а на сервере не тратить места для временных файлов.

Изучим такой вариант. Спасибо!
@VladimirTrue @dr-wh0 что думаете, коллеги?

**Comment by VladimirTrue** [comment]
> > > Да, вы правы. Сначала образ грузится в `/tmp`. Посмотрим как можно улучшить этот момент. Но не забывайте, что Вы тоже можете предложить вариант решения )
> > 
> > А если сделать через NBD (Network Block Device)?
> > 
> > * Подключать ISO в браузере.
> > * Устанавливать соединение с NBD сервером.
> > * Сервер сам решает какие байты принимать.
> > 
> > Тем самым можно подключать ISO в неограниченном размере а на сервере не тратить места для временных файлов.
> 
> Изучим такой вариант. Спасибо!
> @VladimirTrue @dr-wh0 что думаете, коллеги? 

Думаю, что есть смысл рассмотреть этот вариант и поискать способ реализации и интеграции с текущим функционалом. Пока думаю так=)

**Comment by space2pacman** [comment]
Я могу помочь с фронтовой части и интегрировать nbd client так как у меня уже есть наработки.

Но тогда нужно решить как на сервере управлять ISO'шниками. Потому как NBD на стороне сервера представляет ISO'шник как блочное устройство /dev/nbd0, которое готово для чтения.

**Comment by tihon49** [comment]
> Я могу помочь с фронтовой части и интегрировать nbd client так как у меня уже есть наработки.
> 
> Но тогда нужно решить как на сервере управлять ISO'шниками. Потому как NBD на стороне сервера представляет ISO'шник как блочное устройство /dev/nbd0, которое готово для чтения.

Быстрый ресёрч показывает что такой вариант больше подходит к "виртуальной медиа". Нужно провести полноценный ресёрч и понять насколько это будет актуально и удобно для большого кол-ва образов. Какие, потенциальные, точки отказа при этом появятся. Закину эту идею нашему фронтендеру, пусть тоже подумает.
